timemachine/md/minimizer.py

```diff
@@ -349,13 +350,17 @@ def val_and_grad_fn_bfgs(x_local_flattened):
     x_local_0 = x0[local_idxs]
     x_local_0_flat = x_local_0.reshape(-1)
 
-    res = scipy.optimize.minimize(val_and_grad_fn_bfgs, x_local_0_flat, jac=True, options={"disp": True})
+    # use limited memory L-BFGS-B even though we don't have bounds
+    res = scipy.optimize.minimize(
+        val_and_grad_fn_bfgs, x_local_0_flat, method="L-BFGS-B", jac=True, options={"disp": True}
+    )
 
     x_final = x0.copy()
     x_final[local_idxs] = res.x.reshape(x_local_shape)
 
     u_final, _ = val_and_grad_fn(x_final)
 
-    assert u_final < u_0, f"u_0: {u_0:.3f}, u_f: {u_final:.3f}"
+    if u_final >= u_0:
+        warnings.warn(f"Energy minimization with L-BFGS-B failed: u_0: {u_0:.3f}, u_f: {u_final:.3f}")
```

Comment:
would it make sense to return the original coords in this case?

Reply:
probably good just in case